src/authorization-response/Payload.ts

```diff
@@ -19,11 +19,13 @@ export const createResponsePayload = async (
 
   // If state was in request, it must be in response
   const state: string | undefined = await authorizationRequest.getMergedProperty('state');
+  const payload = await authorizationRequest.requestObject.getPayload();
 
   const responsePayload: AuthorizationResponsePayload = {
     ...(responseOpts.accessToken && { access_token: responseOpts.accessToken }),
     ...(responseOpts.tokenType && { token_type: responseOpts.tokenType }),
     ...(responseOpts.refreshToken && { refresh_token: responseOpts.refreshToken }),
+    ...(payload?.nonce && { nonce: payload.nonce}),
```

Comment:
In AuthenticationResponse verify():
```
    const nonce = merged.nonce ?? verifiedIdToken?.payload.nonce ?? oid4vp.nonce;
    ...
    } else if (oid4vp.presentationDefinitions.length > 0 && !nonce) {
      throw Error('Nonce is required when using OID4VP');
    }
```
In my case, if I don't return the nonce it will run into "Nonce is required when using OID4VP"